src/transformStateWithClones.js

```diff
@@ -8,6 +8,35 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const currentState = { ...state };
+  const stateHistory = [];
+
+  for (const obj of actions) {
+    if (obj.type === 'addProperties') {
```

Comment:
Let's use switch case instead of if statement here for better readability

Reply:
there is more than one simple check in the code, is switch case more suitable than the if statement?